app/services/ingest_iati_activities.rb

```diff
@@ -13,15 +14,15 @@ def initialize(delivery_partner:, file_io:)
   def call
     doc = Nokogiri::XML(file_io, nil, "UTF-8")
 
-    legacy_activities = doc.xpath("//iati-activity")
+    legacy_activity_nodes = doc.xpath("//iati-activity")
+    legacy_activity_nodes.each do |legacy_activity_node|
+      legacy_activity = LegacyActivity.new(activity_node_set: legacy_activity_node)
 
-    legacy_activities.each do |legacy_activity|
       ActiveRecord::Base.transaction do
-        new_activity = Activity.new(level: :project, organisation: delivery_partner)
+        new_activity = Activity.find_or_create_by(identifier: legacy_activity.identifier, level: :project, organisation: delivery_partner)
```

Comment:
`new_activity` doesn't strictly make sense in the case when a record is "found" though naming it to activity here and throughout the rest of the code here where `legacy_activity` is used felt more confusing.